packages/react/src/reconciler.ts

```diff
@@ -112,6 +114,24 @@ const reconciler = reactReconciler<
       }
     }
 
+    // eslint-disable-next-line guard-for-in
+    for (const key in updateProps) {
+      const element = updateProps[key] as any;
+      if (!isValidElement(element)) continue;
+
+      let fragmentContainer:
+        | FragmentContainer
+        | undefined = (instance.props as any)[key];
+      if (isFragmentContainer(fragmentContainer)) {
+        delete updateProps[key];
+      } else {
+        fragmentContainer = createFragmentContainer(root);
+        updateProps[key] = fragmentContainer;
+      }
+      const {container} = fragmentContainer;
+      reconciler.updateContainer(element, container, null, () => {});
+    }
```

Comment:
- a, b are expected with current implementation. 
- c: by the time the reconciler receives data, all the context / hooks are resolved. So, it will be fine. 
- d: correct. This is what happens now.

Reply:
I'm not sure I understand what you mean for a-c. Could we codify this behaviour in a test?